src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -709,6 +847,120 @@ biclique xs ys = let sxs = Set.fromList xs
                   in BAM (Map.fromSet (const sys) sxs)
                          (Map.fromSet (const sxs) sys)
 
+-- | The /star/ formed by a center vertex connected to a list of leaves.
+-- Complexity: /O(L log(L))/ time, where /L/ is the length of the given list.
+--
+-- @
+-- star x []     == 'leftVertex' x
+-- star x [y]    == 'edge' x y
+-- star x [x]    == 'edge' x x
+-- star x [y, z] == 'edges' [(x, y), (x, z)]
+-- star x ys     == 'connect' ('leftVertex' x) ('vertices' [] ys)
+-- @
+star :: (Ord a, Ord b) => a -> [b] -> AdjacencyMap a b
+star x ys = overlay (leftVertex x) (edges [ (x, y) | y <- ys ])
+
+-- | The /stars/ formed by overlaying a list of 'star's. An inverse of
+-- 'leftAdjacencyList'.
+-- Complexity: /O(L log(L))/ time, where /L/ is the total size of the input.
+--
+-- @
+-- stars []                      == 'empty'
+-- stars [(x, [])]               == 'leftVertex' x
+-- stars [(x, [y])]              == 'edge' x y
+-- stars [(x, ys)]               == 'star' x ys
+-- stars                         == 'overlays' . 'map' ('uncurry' 'star')
+-- 'overlay' (stars xs) (stars ys) == stars (xs ++ ys)
+-- @
+stars :: (Ord a, Ord b) => [(a, [b])] -> AdjacencyMap a b
+stars xs = overlay (vertices (map fst xs) []) (edges (concat (map sequenceA xs)))
```

Comment:
Isn't this just `overlays . map star`?

Reply:
It is `overlays . map (uncurry star)` as in the example.

It's not clear whether the time complexity remains the same. It uses `Map.unionsWith (Set.union)` inside and their complexity is undocumented. It seems for me that the only way for them to work for `O(L log(L))` in the worst case is if they merge the smaller set/map into the larger.

This actually might be true, but I can't be sure.

The complexity of this function is not critical, but this implementation doesn't seem too complex.

Reply:
Could you compare the performance of simple `overlays . map (uncurry star)` and your version? I always prefer simple solutions to less simple unless there is a good performance reason.

Reply:
If you find that yours is indeed significantly faster (which is not obvious, as you say), then it's worth adding a comment to the implementation, explaining that it is chosen over `overlays . map (uncurry star)` because it gives `Nx` speed up.

Reply:
Well, I didn't realize there's `Map.fromListWith (Set.union)` in `edges`, too.

Reply:
The benchmark says that there is some difference on some tests. The suggested `overlays` implementation works 1.2 times faster on tests where all edges in the input are distinct and the original implementation works 2 times faster on tests where all input stars are equal (with different order of edges, though).

https://gist.github.com/vasalf/9a77006df65ec67be682386fe0a8685d

Reply:
In this case, let's go with the simpler implementation via `overlays . map (uncurry star)`. There are plenty of places like this where we could do some optimisations, but that would derail this PR. After we merge this basic implementation, you are welcome to send more patches for optimising individual functions!